internal/install/application_configuration.go

```diff
@@ -153,13 +153,13 @@ func (ac *ApplicationConfiguration) SetCurrentProfile(name string) {
 // This is mandatory as "elastic-agent-complete" is available since 7.15.0-SNAPSHOT.
 func selectElasticAgentImageName(version, agentBaseImage string) string {
 	if version == "" { // as version is optional and can be empty
-		return elasticAgentLegacyImageName
+		return elasticAgentWolfiImageName
 	}
 
 	v, err := semver.NewVersion(version)
 	if err != nil {
-		logger.Errorf("stack version not in semver format (value: %s): %v", v, err)
-		return elasticAgentLegacyImageName
+		logger.Errorf("stack version not in semver format (value: %s): %v", version, err)
+		return elasticAgentWolfiImageName
 	}
 
 	disableWolfiImages := false
```

Comment:
As it is now, the Elastic Agent image is chosen as:
- If stack >= 8.16.0-SNAPSHOT,
    - by default it is used wolfi image_ `docker.elastic.co/elastic-agent/elastic-agent-wolfi`.
    - if ELASTIC_PACKAGE_DISABLE_ELASTIC_AGENT_WOLFI=true, then it is selected the Elastic Agent image as before. So if stack >= 7.15.0-SNAPSHOT, it will be used the complete image.
- If the configuration file of the system test has `base_image: complete`:
    - It will be used the complete version of the Elastic Agent image if the stack version allows it.
- If the configuration file of the system test has `base_image: systemd`:
    - It will be used the `systemd` version of the Elastic Agent image. That means `docker.elastic.co/elastic-agent/elastic-agent` or `docker.elastic.co/beats/elastic-agent` depending on the stack version.


Would that be ok? Specially if stack version is set to 9.0.0 and the environment variable to disable wolfi is set to true.

Reply:
If the same set of images is going to be available in 9.0 keeping the same logic sounds good to me.